File: mobile/app/src/main/java/com/tomoko/fyntra/data/models/IncidenciaRuta.kt
package com.tomoko.fyntra.data.models

data class IncidenciaRutaCreate(
    val tipo: String, // "averia", "retraso", "cliente_ausente", "otros"
    val descripcion: String,
    val ruta_parada_id: Int? = null // Si es null, es incidencia de ruta; si tiene valor, es de parada
)

data class IncidenciaRutaResponse(
    val id: Int,
    val ruta_id: Int,
    val ruta_parada_id: Int? = null,
    val creador_usuario_id: Int,
    val tipo: String,
    val descripcion: String,
    val creado_en: String,
    val fotos: List<IncidenciaRutaFotoResponse> = emptyList()
)

data class IncidenciaRutaFotoResponse(
    val id: Int,
    val tipo_archivo: String? = null
)

data class EntregaConfirmacion(
    val pedido_id: Int,
    val ruta_id: Int,
    val firma_base64: String? = null,
    val foto_base64: String? = null,
    val observaciones: String? = null
)

File: backend/app/schemas/incidencia_ruta.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime
from app.models.incidencia_ruta import TipoIncidenciaRuta


class IncidenciaRutaFotoResponse(BaseModel):
    id: int
    tipo_archivo: Optional[str] = None

    class Config:
        from_attributes = True


class IncidenciaRutaResponse(BaseModel):
    id: int
    ruta_id: int
    ruta_parada_id: Optional[int] = None
    creador_usuario_id: int
    tipo: TipoIncidenciaRuta
    descripcion: str
    creado_en: datetime
    fotos: List[IncidenciaRutaFotoResponse] = []

    class Config:
        from_attributes = True


class IncidenciaRutaCreate(BaseModel):
    tipo: TipoIncidenciaRuta = Field(..., description="Tipo: averia, retraso, cliente_ausente, otros")
    descripcion: str = Field(..., min_length=1, description="Descripción obligatoria")
    ruta_parada_id: Optional[int] = Field(None, description="ID de la parada de la ruta (opcional). Si no se indica, es incidencia de ruta.")

